### PR TITLE
QA Sprint 156 done

### PR DESCRIPTION
**Pull request by EZheln**
I
Updated tests for Train model wizard
- Jira: ML-5328, ML-5234, ML-5333

II
Updated tests according to requirements
- changed uniqueness validation and for Register artifact, Register dataset
- update messages for Register artifact, Register dataset
- updated mock server functions related to edit and delete tags according to new implementation
- removed the label value validation rules Batch Run wizard
- changed message Tag to Version Tag feature store, feature vector
- added check label validation rules on Create new project pop-up
- Jira: ML-4927, ML-4916, ML-5440, ML-4835

### CODE REVIEW

File: tests/mockServer/mock.js [hunk 1]
```diff
@@ -1165,21 +1165,18 @@ function putTags(req, res){
...
-  
-  if (artifactForUpdate) {
-    if (artifactForUpdate.metadata?.tag || artifactForUpdate.metadata?.tag === ''){
-      artifactForUpdate.metadata.tag = req.params.tag
-    }
-    else if (artifactForUpdate.tag || artifactForUpdate.tag === ''){
-      artifactForUpdate.tag = req.params.tag
-    }
+
+  if (artifactList){
```
Comment (mariana-furyk) [inline tests/mockServer/mock.js]: what if artifactsList length = 0?

File: tests/mockServer/mock.js [hunk 2]
```diff
@@ -1190,20 +1187,19 @@ function putTags(req, res){
 
 function deleteTags(req, res){
   const collectedArtifact = artifacts.artifacts
```
Comment (mariana-furyk) [inline tests/mockServer/mock.js]:
```suggestion
  const collectedArtifacts = artifacts.artifacts
```

File: tests/mockServer/mock.js [hunk 3]
```diff
@@ -1190,20 +1187,19 @@ function putTags(req, res){
...
+        && (artifact.metadata?.uid === req.body.identifiers[0].uid || artifact.metadata?.tree === req.body.identifiers[0].uid || artifact.tree === req.body.identifiers[0].uid) 
+        && (artifact.spec?.db_key === req.body.identifiers[0].key || artifact.db_key === req.body.identifiers[0].key)
+      )
+    )
+  
+  if (collectedArtifact?.length > 1){
+    const artifactByTag = collectedArtifact.find(artifact => (artifact.metadata?.tag === req.params.tag || artifact.tag === req.params.tag))
+    remove(artifacts.artifacts, artifactByTag)
+  }
+  else if (collectedArtifact?.length === 1){
```
Comment (mavdryk) [inline tests/mockServer/mock.js]: else if should be on the previous line

Comment (mavdryk) [inline tests/mockServer/mock.js]: there should a space before opening bracket `{`. Check other places.
Better to enable prettier and eslint

File: tests/mockServer/mock.js [hunk 4]
```diff
@@ -1190,20 +1187,19 @@ function putTags(req, res){
...
-    && ((artifact.spec?.db_key === req.body.identifiers[0].key) 
-    || artifact.db_key === req.body.identifiers[0].key)))
-  if (collectedArtifact) {
-    if (collectedArtifact.metadata?.tag === req.params.tag){
-      collectedArtifact.metadata.tag = ''
-    }
-    else if (collectedArtifact.tag === req.params.tag){
-      collectedArtifact.tag = ''
-    }
+    .filter(artifact => ((artifact.metadata?.project === req.params.project || artifact.project === req.params.project)
```
Comment (mariana-furyk) [inline tests/mockServer/mock.js]: define a const, like artiact = artifact.metadata ?? artifact adn use it where you need

**Review by mavdryk — CHANGES_REQUESTED** [review]

File: tests/mockServer/mock.js [hunk 5]
```diff
@@ -1165,21 +1165,18 @@ function putTags(req, res){
   const tagName = req.params.tag
   const projectName = req.params.project
 
-  const artifactForUpdate = artifacts.artifacts
+  const artifactList = artifacts.artifacts
```
Comment (mariana-furyk) [inline tests/mockServer/mock.js]: use the same name for const ( artifactsList or collectedArtifacts) for the same data

**Review by mariana-furyk — APPROVED** [review]

**Pull request closed without merging**